### PR TITLE
Make Grid List Create Button Respect the Translation of a document type  

### PR DESCRIPTION
**Pull request by IbrahimMNada**
This PR is a fix for Both : 

Fixes https://github.com/umbraco/Umbraco-CMS/issues/17233
Fixes https://github.com/umbraco/Umbraco-CMS/issues/17234

This PR simply checks if the document type label starts with '#' which means that the doctype label is translatable
and then removes the '#' and applies the translations 
Before :
image
image

After:
image
image

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Chore (minor updates related to the tooling or maintenance of the repository, does not impact compiled assets)

## How to test?

1-Create a block list for any doc type with translations
2- Go to block type and Check the CreateButtonLabel
3-you will notice that the label is translated

## Screenshots (if appropriate)
Before :
image
image

After:
image

image

## Checklist

- [ ] If my change requires a change to the documentation, I have updated the documentation in this pull request.
- [x] I have read the **[CONTRIBUTING](<(https://github.com/umbraco/Umbraco.CMS.Backoffice/blob/main/.github/CONTRIBUTING.md)>)** document.
- [] I have added tests to cover my changes.

### CODE REVIEW

File: src/packages/block/block/workspace/block-workspace-editor.element.ts [hunk 1]
```diff
@@ -19,7 +19,7 @@ export class UmbBlockWorkspaceEditorElement extends UmbLitElement {
 					context.content.structure.ownerContentTypeObservablePart((contentType) => contentType?.name),
 				]),
 				([isNew, name]) => {
-					this._headline = this.localize.term(isNew ? 'general_add' : 'general_edit') + ' ' + name;
+					this._headline = this.localize.term(isNew ? 'general_add' : 'general_edit') + ' ' +  this.localize.term(name!);
```
Comment (iOvergaard) [inline src/packages/block/block/workspace/block-workspace-editor.element.ts:22]:
```suggestion
					this._headline = this.localize.term(isNew ? 'general_add' : 'general_edit') + ' ' +  this.localize.string(name!);
```

File: src/packages/block/block/workspace/views/edit/block-workspace-view-edit.element.ts [hunk 2]
```diff
@@ -149,7 +149,7 @@ export class UmbBlockWorkspaceViewEditElement extends UmbLitElement implements U
 								(tab) => {
 									const path = this._routerPath + '/tab/' + encodeFolderName(tab.name || '');
 									return html`<uui-tab label=${tab.name ?? 'Unnamed'} .active=${path === this._activePath} href=${path}
-										>${tab.name}</uui-tab
+										>${this.localize.term(tab.name)}</uui-tab
```
Comment (iOvergaard) [inline src/packages/block/block/workspace/views/edit/block-workspace-view-edit.element.ts:152]:
```suggestion
										>${this.localize.string(tab.name)}</uui-tab
```

**Review by iOvergaard — CHANGES_REQUESTED** [review]
Made a few suggestions to use `string` rather than `term` as that already covers the proposed use case.

**Comment by IbrahimMNada** [comment]
I Will work on them Now

**Comment by IbrahimMNada** [comment]
changes done and tested all looks good

**Review by iOvergaard — APPROVED** [review]
I have changed the `string` function to accept unknown values, so that it can validate them at runtime. This allows for a better DX experience as you do not have to code defensively on the values passed to it. I have added tests to ensure that this scenario works. I have also fixed a few Sonarcloud issues that arose on unrelated code due to the fact, that it checks changed files only.

**Pull request merged**